benchmark/cublas/blas1/iamax.cpp

```diff
@@ -0,0 +1,169 @@
+/**************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename iamax.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+#include "common/common_utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(int size) {
+  std::ostringstream str{};
+  str << "BM_Iamax<" << blas_benchmark::utils::get_type_name<scalar_t>();
+  str << ">/" << size;
+  return str.str();
+}
+
+template <typename scalar_t, typename... args_t>
+static inline void cublas_routine(args_t&&... args) {
+  if constexpr (std::is_same_v<scalar_t, float>)
+    CUBLAS_CHECK(cublasIsamax(std::forward<args_t>(args)...));
+  else if constexpr (std::is_same_v<scalar_t, double>)
+    CUBLAS_CHECK(cublasIdamax(std::forward(args)...));
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, cublasHandle_t* cuda_handle_ptr, index_t size,
+         bool* success) {
+  // Google-benchmark counters are double.
+  double size_d = static_cast<double>(size);
+  state.counters["size"] = size_d;
+  state.counters["n_fl_ops"] = 2 * size_d;
+  state.counters["bytes_processed"] = size_d * sizeof(scalar_t);
+
+  cublasHandle_t& cuda_handle = *cuda_handle_ptr;
+
+  using tuple_scalar_t = blas::IndexValueTuple<index_t, scalar_t>;
+
+  // Create data
+  std::vector<scalar_t> v1 = blas_benchmark::utils::random_data<scalar_t>(size);
+  tuple_scalar_t out{-1, 0};
+
+  // This will clamp the values to what scalar_t can represent
+  std::transform(std::begin(v1), std::end(v1), std::begin(v1), [](scalar_t v) {
+    return utils::clamp_to_limits<scalar_t>(v);
+  });
+
+  scalar_t* d_inx = nullptr;
+  CUDA_CHECK(cudaMalloc(&d_inx, size * sizeof(scalar_t)));
+  CUDA_CHECK(cudaMemcpy(d_inx, v1.data(), size * sizeof(scalar_t),
+                        cudaMemcpyHostToDevice));
+
+  index_t* d_out = nullptr;
+  CUDA_CHECK(cudaMalloc(&d_out, sizeof(index_t)));
+  CUDA_CHECK(cudaDeviceSynchronize());
+
+  CUBLAS_CHECK(
+      cublasSetPointerMode_v2(cuda_handle, CUBLAS_POINTER_MODE_DEVICE));
```

Comment:
Do you really need `cublasSetPointerMode_v2` in place of `cublasSetPointerMode`?

Reply:
yes it is. Almost every cublas call in reality is a ```define``` like this 
```
#define cublas_operator cublas_operator_v2
```
My lsp doens't give me hint if use the first one and, like in this case, I forgot to remove the _v2. I should fix it for consistency...